### PR TITLE
Fix doubly linked list algorithm

### PR DESCRIPTION
**Pull request by TheSuperNoob**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by TheSuperNoob** [comment]
This also includes adding the doctests from #2052

File: data_structures/linked_list/doubly_linked_list.py [hunk 1]
```diff
@@ -9,76 +9,109 @@
...
+            self.head = None
+            self.tail = None
+
+        return head_data
+
+    def insert_at_tail(self, data):
+        new_node = Node(data)
+        self.tail.next = new_node
+        new_node.previous = self.tail
+        self.tail = new_node
```
Comment (mquevill) [inline data_structures/linked_list/doubly_linked_list.py:60]: This does not work if the list is empty. Copying your logic from `insert_at_head()`:
```suggestion
    def insert_at_tail(self, data):
        new_node = Node(data)
        if self.is_empty:
            self.head = new_node
            self.tail = new_node
        else:
            self.tail.next = new_node
            new_node.previous = self.tail
            self.tail = new_node
```

Comment (mquevill) [reply data_structures/linked_list/doubly_linked_list.py:60]: This was raised in #2112.

File: data_structures/linked_list/doubly_linked_list.py [hunk 2]
```diff
@@ -9,76 +9,109 @@
...
+            self.delete_tail()
 
         else:  # Before: 1 <--> 2(current) <--> 3
             current.previous.next = current.next  # 1 --> 3
             current.next.previous = current.previous  # 1 <--> 3
+        return data
 
-    def isEmpty(self):  # Will return True if the list is empty
+    @property
+    def is_empty(self):  # Will return True if the list is empty
```
Comment (cclauss) [inline data_structures/linked_list/doubly_linked_list.py:97]:
```suggestion
    def is_empty(self):  # return True if the list is empty
```

File: data_structures/linked_list/doubly_linked_list.py [hunk 3]
```diff
@@ -9,76 +9,109 @@
...
         return self.head is None
 
     def display(self):  # Prints contents of the list
```
Comment (cclauss) [inline data_structures/linked_list/doubly_linked_list.py:100]: Could we please define `__str__()` instead and let the caller print or log or whatever.

**Pull request merged**